tests/test_dag_extension.py

```diff
@@ -0,0 +1,104 @@
+from marquez.airflow import MarquezDag
+from unittest.mock import Mock, patch, create_autospec
+from datetime import datetime
+from airflow.utils.state import State
+import airflow.models
+import pytest
+from croniter import croniter
+import pendulum
+import marquez.utils
+from marquez_client.marquez import MarquezClient
+
+
+class Context:
+    location = 'github://test_dag_location'
+    dag_id = 'test-dag-1'
+    namespace = 'test-namespace-1'
+    data_inputs = ["s3://data_input_1", "s3://data_input_2"]
+    data_outputs = ["s3://some_output_data"]
+    owner = 'test_owner'
+    description = 'this is a test DAG'
+    airflow_run_id = 'airflow_run_id_123456'
+    mqz_run_id = '71d29487-0b54-4ae1-9295-efd87f190c57'
+    start_date = datetime(2019, 1, 31, 0, 0, 0)
+    execution_date = datetime(2019, 2, 2, 0, 0, 0)
+    schedule_interval = '*/10 * * * *'
+
+    dag = None
+
+    def __init__(self):
+        self.dag = MarquezDag(
+            self.dag_id,
+            schedule_interval=self.schedule_interval,
+            default_args={'mqz_namespace': self.namespace,
+                          'mqz_location': self.location,
+                          'mqz_input_datasets': self.data_inputs,
+                          'mqz_output_datasets': self.data_outputs,
+                          'owner': self.owner,
+                          'depends_on_past': False,
+                          'start_date': self.start_date},
+            description=self.description)
+
+
+@pytest.fixture(scope="module")
+def context():
+    return Context()
+
+
+@patch.object(airflow.models.DAG, 'create_dagrun')
+@patch.object(marquez.utils.JobIdMapping, 'set')
+def test_create_dagrun(mock_set, mock_dag_run, context):
+
+    dag = context.dag
+    mock_mqz_client = make_mock_mqz_client(context.mqz_run_id)
+    dag._mqz_client = mock_mqz_client  # Use a mock marquez-python client
+    mock_dag_run.return_value = make_mock_airflow_jobrun(dag.dag_id, context.airflow_run_id)
+
+    # trigger an airflow DagRun
+    dag.create_dagrun(state=State.RUNNING, run_id=context.airflow_run_id, execution_date=context.execution_date)
```

Comment:
nice that we're re-using airflow's state instead of re-defining our own!

Reply:
We'll want to define are own run state enum and map them to `airflow` states. Not sure if it's a direct 1:1, but this will ensure we cab add to the set of known `marquez` run states.

Reply:
btw, this is airflow's runState argument, nothing to do with Marquez

Reply:
Oh, sorry! But any thoughts on defining our own `marquez` run state enum?

Reply:
Hmm, I'm thinking the enum should be provided by the `marquez` client, so let's revisit this idea later

Reply:
I also agree the marquez-client should have an enum, but we should discuss that on that project. But currently marquez-client has methods for each jobrun state change.